### PR TITLE
add option to subtract template at trigger time

### PR DESCRIPTION
**Pull request by ahnitz**

### CODE REVIEW

File: bin/pycbc_single_template [hunk 1]
```diff
@@ -317,7 +329,15 @@ with ctx:
         snr, corr, norm = filter.matched_filter_core(template, stilde, 
                                     psd=stilde.psd,
                                     low_frequency_cutoff=flow)
-        snr *= norm
+        snr *= norm 
+                                    
+        if opt.subtract_template:
+            stilde = subtract_template(stilde, template,
+                                snr, opt.trigger_time, opt.low_frequency_cutoff)
```
Comment (spxiwh) [inline bin/pycbc_single_template]: Just double checking. I don't see the recovered phase here, but I assume snr at this point is complex and then you do a complex multiplication above to ensure that the template is subtracted at the correct phase?

Comment (ahnitz) [inline bin/pycbc_single_template]: Yes, that's right. I use the given trigger time to pick off the righ complex SNR value. This is used to scale both the distance and the phase correctly.

**Comment by spxiwh** [comment]
Landscape is failing, but it seems unrelated to this. Are these things that have *been* fixed, or things that slipped through cracks when landscape did not run?

Anyway, maybe best to rebase.

**Comment by ahnitz** [comment]
Well, I rebased...

**Pull request merged**